extensions/metadata_exchange/plugin.h

```diff
@@ -89,6 +89,8 @@ class PluginContext : public Context {
   inline StringView nodeId() { return rootContext()->nodeId(); }
 
   ::Wasm::Common::TrafficDirection direction_;
+  bool metadata_received_{true};
```

Comment:
HTTP spec allows response data before the request data, so we have to opportunistically guess it's received then.